### DISCUSSION TITLE
I had already setup using steamctl how to migrate to steamguard-cli?

### DISCUSSION THREAD

**Discussion by brunoais** [dyc3/steamguard-cli#399] [Q&A] [answered]
I had already setup using steamctl. I know where the json file is (it is encrypted by a custom wrapper I did around the tool) and I can read it.
How to migrate to steamguard-cli?

(I searched for `steamctl` in the discussions and I couldn't find it)

-----------

Forgot to add how the file looks like:
`username.json`
```
{
    "account_name": "username",
    "identity_secret": "abcQ=",
    "phone_number_hint": "3423",
    "revocation_code": "R7511",
    "secret_1": "lettersDigitsAndCapiltas/lettersDigitsAndCapiltas=",
    "serial_number": "DigitsDigitsDigits",
    "server_time": "1893456000",
    "shared_secret": "lettersDigitsAndCapiltas/lettersDigitsAndCapiltas=",
    "status": 1,
    "token_gid": "hexnumber",
    "uri": "otpauth://totp/Steam:username?secret=LETTERSDIGITSINCAPITALS&issuer=Steam"
}
```
All values are placeholders of the real value.

**Comment by dyc3** [comment]
These threads should be useful:

https://github.com/dyc3/steamguard-cli/discussions/394
https://github.com/dyc3/steamguard-cli/discussions/388

**Reply by brunoais** [reply]
I'm missing `device_id` and `steam_id`. I think I can easily get `steam_id` assuming it's my userid. However, I don't know how to get the `device_id`.
Any idea? Is it mandatory?

**Reply by dyc3** [reply]
yeah, `steam_id` is the integer version of your steam id. `device_id` is generated here:
https://github.com/dyc3/steamguard-cli/blob/96a30c615015606eada9102ea33bd7466b45a537/steamguard/src/accountlinker.rs#L281-L283
IIRC, the content doesn't really matter. It's just for correlating the mafiles to the keys that are stored on valve's side.

You can also use this for reference: https://github.com/dyc3/steamguard-cli/blob/master/src/fixtures/maFiles/manifest-v1/1-account/1234.maFile

**Comment by brunoais** [comment, marked as answer]
With the guidance of dyc3 responses, I made it happen.
I found my steam id at: https://store.steampowered.com/account/
I generated a uuid at: https://www.uuidgenerator.net/

Then created: `/path/to/123456789.maFile` (dummy steam id)
Wrote steamctl's json to it.
Then, based on https://github.com/dyc3/steamguard-cli/blob/master/src/fixtures/maFiles/manifest-v1/1-account/1234.maFile, for reference, I added:
```
    "steam_id": 123456789,
    "device_id": "android:d406ed18-8039-40c3-a954-5fbbcda9edd0"
```
to the file, just before the `}`

Then, based on https://github.com/dyc3/steamguard-cli/blob/master/src/fixtures/maFiles/manifest-v1/1-account/manifest.json, for reference, I inserted:
```
{"filename":"123456789.maFile","steam_id":123456789,"account_name":"username","encryption":null}
```
between the `[]` (which is after `"entries":`)

Then it just works without anything else.

Thank you for the guidance, @dyc3

**Comment by brunoais** [comment]
For the ones wondering about security, I run steamctl and steamguard-cli under a separate user that cannot be logged in into and the directories to where the files are written to are only writable by that user and root.
I use `sudo -su steamguard{,}` to run it.
Also, the files are in an encrypted drive, so it's also unfeasible to try accessing with an alternative OS.
